extensions/ql-vscode/src/queryRunner.ts

```diff
@@ -54,4 +97,53 @@ export abstract class QueryRunner {
   ): Promise<void>;
 
   abstract clearPackCache(): Promise<void>;
+
+  /**
+   * Create a `CoreQueryRun` object. This creates an object whose `evaluate()` function can be
+   * called to actually evaluate the query. The returned object also contains information about the
+   * query evaluation that is known even before evaluation starts, including the unique ID of the
+   * evaluation and the path to its output directory.
+   */
+  public createQueryRun(
+    dbPath: string,
+    query: CoreQueryTarget,
+    generateEvalLog: boolean,
+    additionalPacks: string[],
+    queryStorageDir: string,
+    id: string | undefined,
```

Comment:
Minor: is this cleaner than what you have? And then you can remove `actualId` below.

```suggestion
    id: string = `${basename(query.queryPath)}-${nanoid()}`,
```